src/CommunityToolkit.Aspire.Hosting.PowerShell/PowerShellRunspacePoolResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Aspire.Hosting;
+using Aspire.Hosting.ApplicationModel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics.CodeAnalysis;
+using System.Management.Automation;
+
+namespace CommunityToolkit.Aspire.Hosting.PowerShell;
+
+/// <summary>
+/// Extensions for the PowerShellRunspacePoolResourceBuilder.
+/// </summary>
+public static class PowerShellRunspacePoolResourceBuilderExtensions
+{
+    /// <summary>
+    /// Adds a PowerShell script resource to the distributed application.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="name"></param>
+    /// <param name="script"></param>
+    /// <returns></returns>
+    public static IResourceBuilder<PowerShellScriptResource> AddScript(
+        this IResourceBuilder<PowerShellRunspacePoolResource> builder,
+        [ResourceName] string name,
+        [StringSyntax("PowerShell")] string script)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(name);
+        ArgumentException.ThrowIfNullOrWhiteSpace(script);
+
+        // parse to force an early exception if the script is invalid
+        var scriptBlock = ScriptBlock.Create(script);
+
+        var scriptResource = new PowerShellScriptResource(name, scriptBlock, builder.Resource);
+
+        builder.ApplicationBuilder.Eventing.Subscribe<InitializeResourceEvent>(scriptResource, (e, ct) =>
+        {
+            var loggerService = e.Services.GetRequiredService<ResourceLoggerService>();
+            var notificationService = e.Services.GetRequiredService<ResourceNotificationService>();
+
+            var scriptName = scriptResource.Name;
+            var scriptLogger = loggerService.GetLogger(scriptName);
+            try
+            {
+                // TODO: capture script streams and log them
+                scriptLogger.LogInformation("Starting script '{ScriptName}'", scriptName);
+
+                _ = notificationService
+                    .WaitForDependenciesAsync(scriptResource, ct)
+                    .ContinueWith(
+                        async (_) => await scriptResource.StartAsync(scriptLogger, notificationService, ct),
+                        ct);
+            }
+            catch (Exception ex)
+            {
+                scriptLogger.LogError(ex, "Failed to start script '{ScriptName}'", scriptName);
+            }
+
+            return Task.CompletedTask;
+        });
+
+        return builder.ApplicationBuilder
+            .AddResource(scriptResource)
+            .WaitFor(builder) // wait for pool resource
```

Comment:
You don't need to do this if the `PowerShellScriptResource` is defined as a `IResourceWithParent` (see other comment) as this resource will know it's a child resource and it'll wait until the parent is ready.

Reply:
See other comment. Won't fix.